### PR TITLE
KOGITO-7248: Patch PR for KOGITO-6996 with doc improvements in parallelism guide

### PR DESCRIPTION
**Pull request by hmanwani-rh**
**JIRA:** https://issues.redhat.com/browse/KOGITO-7248

**Description:** Patch PR for KOGITO-6996 with doc improvements in parallelism guide

https://github.com/kiegroup/kogito-docs/pull/77

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category
- [x] The index.adoc file has a card to this guide in the proper category, with a meaningful description

### CODE REVIEW

**Comment by hmanwani-rh** [comment]
@hbelmiro Please review the changes applied in this PR and let me know if the changes look good.

**Review by hbelmiro — CHANGES_REQUESTED** [review]
Thanks for your improvements @hmanwani-rh.
I left a few comments.

File: serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
+This document describes how you can run parallel tasks in Serverless Workflow. 
+
+The testing procedure described in this document is based on the link:{kogito_sw_examples_url}/serverless-workflow-service-calls-quarkus[`serverless-workflow-service-calls-quarkus`] example application in GitHub repository.
 
 [WARNING]
 ====
-Kogito Serverless Workflow serialize parallel execution. Therefore, "parallel" does not mean simultaneous execution. It means there is no logical dependency between branches. An inactive branch can start/resume execution without waiting completion of the active branch if the latter suspends execution (for example, wait for an event reception).
+Serverless Workflow serializes the execution of parallel tasks. Therefore, the word "parallel" does not indicate simultaneous execution, but it means that there is no logical dependency between the execution of branches. An inactive branch can start or resume the execution of a task without waiting for an active branch to be completed, in case the latter suspends the execution (for example, wait for an event reception).
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc:13]: Done, added Kogito and removed Serverless Workflow.

File: serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc [hunk 2]
```diff
@@ -96,41 +96,45 @@ include::../../pages/_common-content/getting-started-requirement.adoc[]
...
+* A parallel workflow is created. 
++
+For more information, see <<proc-parallel-creating-the-workflow, Creating a parallel Serverless Workflow>>.
 
-* The workflow application is up and running.
-You can start it in dev mode by running:
+* The Serverless Workflow application is up and running.
 +
 --
+You can start the Serverless Workflow application in developer mode by using the following command:
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc:126]: Done

File: serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc [hunk 3]
```diff
@@ -96,41 +96,45 @@ include::../../pages/_common-content/getting-started-requirement.adoc[]
...
+<3> Function that concatenates the string `C`.
+<4> Defines the type of the state as `parallel`.
+<5> Defines the branches that run in parallel.
+<6> Defines the function that runs in `branchA`.
+<7> Defines the function that runs in `branchB`.
+<8> Defines the function that runs in `branchC`.
 <9> Defines the completion type of the parallel state as `allOf`.
-This means that the parallel state will be completed when all the branches are completed.
 
+This means that the parallel state is completed when all the branches are completed.
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/core/working-with-parallelism.adoc:109]: Done

**Comment by hmanwani-rh** [comment]
> Thanks for your improvements @hmanwani-rh. I left a few comments.

@hbelmiro Thanks for reviewing the changes and sharing your comments. I've incorporated the changes as suggested.

**Review by hbelmiro — APPROVED** [review]

**Pull request merged**